### PR TITLE
Keep enum configuration backwards compatible

### PR DESCRIPTION
**Pull request by apc999**
Fix https://alluxio.atlassian.net/browse/ALLUXIO-3402

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 1]
```diff
@@ -304,14 +304,12 @@ public boolean getBoolean(PropertyKey key) {
...
     String rawValue = get(key);
-
     return ConfigurationUtils.parseAsList(rawValue, delimiter);
   }
 
   @Override
   public <T extends Enum<T>> T getEnum(PropertyKey key, Class<T> enumType) {
-    String rawValue = get(key);
-
+    String rawValue = get(key).toUpperCase();
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:312]: Not sure is there some corner case that the member of T enumType is lowerCase or mix lowerCase and upperCase?

Comment (apc999) [reply core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:312]: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

enumType should be in upper case as they are constant to follow Java convention. 
Also in here,  it only works for a few enum type propertyKey. I believe they are all in upper case

Comment (maobaolong) [reply core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:312]: OK, sounds good.

**Review by maobaolong — APPROVED** [review]
LGTM, thanks for the backwards compatible improvement.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**